### DISCUSSION TITLE
How to configure Selectize/TomSelect?

### DISCUSSION THREAD

**Discussion by Actionb** [jrief/django-formset#60] [Q&A]
How can I pass [options/settings](https://tom-select.js.org/docs/) on to the TomSelect constructor for customization? 

For example, when I want to use the [dropdown header plugin](https://tom-select.js.org/plugins/dropdown-header/)  or when I want to [override how options are rendered](https://tom-select.js.org/docs/#render-templates)?

**Comment by jrief** [comment]
I could add an extra attributes to the webcomponent implementing the Tom-Select widget, for instance `<select is="django-selectize" dropdown-header="My Title" … >`. This value then could be passed in by the Django widget's attributes.

As for how options are rendered, this could be implemented by additional `<template>…</template>` elements in a template overriding `formset/templates/formset/default/widgets/selectize.html`.

**Comment by Actionb** [comment]
Would it be possible to allow passing the options for TomSelect in the widget itself like this: https://github.com/jrief/django-formset/commit/6278973a298b03c4dddd9528ab80000947120209?

The above disables match highlighting within the dropdown menu as an example. It's not exactly pretty, but it is flexible.

**Comment by Actionb** [comment]
On second thought, this is not robust enough, since you can't pass functions to the constructor this way.

**Comment by jrief** [comment]
I would prefer a list of dedicated options inside a special attribute, passed to the web component implementing the widget. "highlight" could be one of them. This would allow me to reimplement these options in a possible future alternative library.
